src/Float/Extra.elm

```diff
@@ -0,0 +1,368 @@
+    format =
+        Float.Extra.toFixedSignificantDigits 3
+            |> Float.Extra.boundaryValuesAsUnicode
+
+    format (0 / 0) --> "∅"
+    format (1 / 0) --> "∞"
+    format (-1 / 0) --> "-∞"
+    format (1 / 3) -> "0.333"
+
+Of course using this is unsuitable for when you want the numbers to be machine readable.
+
+-}
+boundaryValuesAsUnicode : (Float -> String) -> Float -> String
+boundaryValuesAsUnicode formatter value =
+    if isNaN value then
+        "∅"
+
+    else if isInfinite value then
+        sign value ++ "∞"
+
+    else
+        formatter value
+
+
+
+-- aboutEqual
+
+
+{-| Comparing Floats with `==` is usually wrong, unless you basically care for reference equality, since floating point
+numbers often have small precision drift.
+
+    0.1 + 0.2 == 0.3 --> False
+
+This function implements an approximation where we are asking - are these values close enough that we can consider their difference to be
+due to floating point drift rather than a result of meaningful difference in calculation?
+
+    (0.1 + 0.2) |> Float.Extra.aboutEqual 0.3 --> True
+
+Note: this is unlikely to be appropriate if you are performing computations much smaller than one.
+
+    (0.00001 + 0.00002) |> Float.Extra.aboutEqual 0.00003 --> True
+
+This value handles Infinity and NaN like so:
+
+    (1 / 0) |> Float.Extra.aboutEqual (100 / 0) --> True
+
+    (0 / 0) |> Float.Extra.aboutEqual (0 / 0) --> False
+
+-}
+aboutEqual : Float -> Float -> Bool
+aboutEqual a b =
+    if isInfinite a then
+        isInfinite b
+
+    else if isInfinite b then
+        False
+
+    else
+        abs (a - b) <= 1.0e-5 + 1.0e-8 * abs a
+
+
+
+-- Range
+
+
+{-| Returns a List containing an arithmetic progression, similar to the Python
+built-in range.
+
+Takes a `start`, `stop` and `step` argument. The stop value is exclusive; it is not
+included in the result. If `step` is positive, the last element is the largest
+`start + i * step` less than `stop`; if `step` is negative, the last element is
+the smallest `start + i * step` greater than `stop`. If the returned list would
+contain an infinite number of values, an empty range is returned.
+
+The arguments are not required to be whole numbers; however, the results are more
+predictable if they are.
+
+Differences from [List.range from the standard library](https://package.elm-lang.org/packages/elm/core/latest/List#range):
+
+  - `List.range` is inclusive, meaning that the stop value will be included in the result
+  - `List.range` supports `Int`, whereas this uses `Float`
+  - `List.range` supports only increasing intervals (i.e. `List.range 3 1 == []` vs. `range 3 1 -1 == [3, 2]`)
+  - `List.range` doesn't allow for specifying the step value
+
+-}
+range : Float -> Float -> Float -> List Float
+range start stop step =
+    let
+        n =
+            (stop - start)
+                / step
+                |> ceiling
+                -- get rid of NaN
+                |> Bitwise.or 0
+                |> max 0
```

Comment:
Wouldn't it be simpler to just check that `step /= 0`? Cause if `start` or `stop` is NaN then... `n` is `NaN` and `i >= 0` is `False` and nevermind it's incredibly bad to rely on such a subtle detail